### PR TITLE
feature/basic endpoints and validation

### PR DESCRIPTION
**Pull request by jprittie**
- Implement `GET /posts`, `POST /posts`, `PUT /posts/:id` and `DELETE /posts:id` endpoints 
- Validate request body with `express-validator` 
- Add `validationErrorHandler` middleware
- Create Swagger docs with `swagger-ui-express`
- Add `prettier` to project

### CODE REVIEW

**Comment by jprittie** [comment]
@ysim

File: src/routes/posts/posts.controller.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+const listPosts = (req, res) => {
+  return res.json({
+    data: postsData
+  });
+};
+
+/* POST /posts controller */
+const createPost = async (req, res) => {
+  // Throw 422 if validation errors are found
+  const errors = validationResult(req);
```
Comment (ysim) [inline src/routes/posts/posts.controller.js]: Hmm, has the request not already been validated using `validationErrorHandler`?

File: src/routes/posts/posts.controller.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      })
+    );
+    return next(error)
+  }
+
+  // Throw 422 if validation errors are found
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(422).json({ errors: errors.array() });
+  }
```
Comment (ysim) [inline src/routes/posts/posts.controller.js]: Ditto comment above on validation

File: src/routes/posts/posts.controller.js (hunk 1 above)
Comment (jprittie) [reply src/routes/posts/posts.controller.js]: @ysim Yes, you're right — it was applied in the router! Thanks for catching that.

**Pull request merged**